src/sentry/api/endpoints/organization_spans_fields.py

```diff
@@ -413,10 +413,11 @@ def __init__(
     def resolve_attribute_key(self, key: str, snuba_params: SnubaParams) -> AttributeKey | None:
         resolver = SearchResolver(params=snuba_params, config=SearchResolverConfig())
         resolved, _ = resolver.resolve_attribute(key)
-        proto = resolved.proto_definition
-        if not isinstance(proto, AttributeKey):
+        if resolved.search_type != "string" or not isinstance(
+            resolved.proto_definition, AttributeKey
+        ):
```

Comment:
heads up this change; https://github.com/getsentry/sentry/pull/80829/files#diff-ca53615e4a5d771b967f49cd366caf62d188212ece593a146cf4b31de578015aR67-R79 
will mean we don't need to do the `isinstance` part of the check anymore